TCAT/Controllers/HomeViewController.swift

```diff
@@ -129,29 +127,26 @@ class HomeViewController: UIViewController {
     }
 
     override var preferredStatusBarStyle: UIStatusBarStyle {
-        return isBannerShown ? .lightContent : .default
+        return banner != nil ? .lightContent : .default
     }
 
-    @objc func reachabilityChanged(note: Notification) {
-        if let reachability = note.object as? Reachability {
+    @objc func reachabilityChanged(_ notification: Notification) {
+        if let reachability = notification.object as? Reachability {
+            
+            // Dismiss current banner, if any
+            banner?.dismiss()
+            banner = nil
+            
             switch reachability.connection {
             case .none:
-                if !isBannerShown {
-                    banner = StatusBarNotificationBanner(title: Constants.Banner.noInternetConnection, style: .danger)
-                    banner!.autoDismiss = false
-                    banner!.show(queuePosition: .front, on: navigationController)
-                    isBannerShown = true
-                }
+                banner = StatusBarNotificationBanner(title: Constants.Banner.noInternetConnection, style: .danger)
+                banner!.autoDismiss = false
```

Comment:
I know the banner exists, but can we please do an `if let banner = banner {...}` Would appreciate it!

Reply:
@AAAstorga I reset the banner to be nil in this function to reset it properly. Would it be better to just use `?` for lines 143 and such? `if let` would be pretty redundant, I'd have to set the variable and then check if it's not nil.